gnocchi/opts.py

```diff
@@ -172,7 +171,16 @@ def list_opts():
                        default=10000,
                        min=1,
                        help="Number of metrics that should be deleted "
-                       "simultaneously by one janitor.")
+                            "simultaneously by one janitor."),
+            cfg.IntOpt('metric_inactive_after',
```

Comment:
there's some ambiguity with what time ended_at value is set to.
- the ended_at value is dependent on when the janitor runs. this will lag actual metric_inactive_after value by an unknown amount.
- we update last_measure_timestamp based on when metricd worker processes metric. this means this will also lag an unknown amount when actual last measure was captured and/or sent/received by gnocchi.

not sure but is it better to mark ended_at with latest last_measure_timestamp + metric_inactive_after or does precision not really matter?

Reply:
yes, there is. I mean, it is something that does not need to be precise. What matter is not the seconds when the metric is considered as inactive, but that all of them are inactivated. 

Also, for the resource, if we want to do "last_measure_timestamp + metric_inactive_after", from which metric we take the "last_measure_timestamp"? A resource can have multiple. We can always take the last one that became inactive, and so on. However, I do not see what sort of benefits this would bring.

Reply:
i had thought historically, gnocchi has strayed from handling time as of "now"/wall time. that said, i can't find any reference to this explicitly so i guess it doesn't matter.

if you don't care about precision, i'm ok with it. i can't imagine it'd be controversial to change this behaviour in future if that is preference.